llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2263,6 +2263,11 @@ static bool isKnownNonNullFromDominatingCondition(const Value *V,
         return true;
     }
 
+    if ((match(U, m_UDiv(m_Value(), m_Specific(V))) ||
+         match(U, m_SDiv(m_Value(), m_Specific(V)))) &&
```

Comment:
```suggestion
    if (match(U, m_IDiv(m_Value(), m_Specific(V))) &&
```

Reply:
@dtcxzyw I think its fine to keep this PR to just IDiv (I'm mainly interested in closing the issue), the others can be implemented separately.